Eatery/Controllers/Eateries/Campus/CampusEateryMealTableViewController.swift

```diff
@@ -88,80 +106,90 @@ class CampusEateryMealTableViewController: UITableViewController {
         ) as! MealStationTableViewCell
 
         cell.titleLabel.text = "No menu available"
-        cell.contentLabel.attributedText = NSAttributedString(string: "")
 
         return cell
     }
 
     /// Create a table view cell when there is a single dining station in the menu
     private func menuItemCell(in tableView: UITableView, forRowAt indexPath: IndexPath) -> UITableViewCell {
-        guard let menu = menu, let station = menu.data.first else {
+        guard let menuData = menu?.data else {
             return emptyMenuCell(in: tableView, forRowAt: indexPath)
         }
-
+        guard let menu = menu else {
+            return emptyMenuCell(in: tableView, forRowAt: indexPath)
+        }
+        let station = menu.data[menuData.index(menuData.startIndex, offsetBy: indexPath.section)]
         let name = station.value[indexPath.row].name
 
         let cell = tableView.dequeueReusableCell(withIdentifier: "MealItem", for: indexPath) as! MealItemTableViewCell
         cell.nameLabel.text = name
         return cell
     }
 
-    /// Create a table view cell when there are multiple dining stations in the menu
+    /// Create a table view cell header when there are multiple dining stations in the menu
     private func diningStationsCell(in tableView: UITableView, forRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(
-            withIdentifier: "MealStation",
+            withIdentifier: "MealStationItem",
             for: indexPath
-        ) as! MealStationTableViewCell
+        ) as! MealStationItemTableViewCell
 
         guard let menu = menu else {
             return emptyMenuCell(in: tableView, forRowAt: indexPath)
         }
 
         let sortedMenu = Sort.sortMenu(menu.data.map { ($0, $1) })
-        let stationTitles = sortedMenu.map { $0.0 }
-
-        // set title
-        let possibleTitle = stationTitles[indexPath.row]
-        if possibleTitle == "General" {
-            cell.titleLabel.text = ""
-            cell.titleCollapsed = true
-        } else {
-            cell.titleLabel.text = possibleTitle
-            cell.titleCollapsed = false
-        }
-
-        // set content
-        let menuItems = sortedMenu[indexPath.row].1
-        let names = menuItems.map { item -> NSMutableAttributedString in
-            if item.healthy {
-                return NSMutableAttributedString(string: "\(item.name.trim()) ")
-                    .appendImage(UIImage(named: "appleIcon")!, yOffset: -1.5)
-            } else {
-                return NSMutableAttributedString(string: item.name)
-            }
-        }
-
         let font = UIFont.systemFont(ofSize: 14)
         let paragraphStyle = NSMutableParagraphStyle()
         paragraphStyle.paragraphSpacing = 0.25 * font.lineHeight
-        let content = NSMutableAttributedString(
-            string: "",
+
+        let menuItem = sortedMenu[indexPath.section].1[indexPath.item-1]
```

Comment:
```suggestion
        let menuItem = sortedMenu[indexPath.section].1[indexPath.item - 1]
```
Spaces around `-`. Also, be careful when you index into this array. Only do this if you are 100% certain that the index will be in bounds.